packages/mobile/src/qrcode/utils.ts

```diff
@@ -166,3 +175,29 @@ export function* handleBarcode(
 
   yield call(handleSendPaymentData, qrData, cachedRecipient, isOutgoingPaymentRequest, true)
 }
+type PaymentDeepLinkHandlers = {
+  [key in PaymentDeepLinkHandler]: (barcode: QrCode) => Generator
+}
+
+const paymentDeepLinkHandlers: PaymentDeepLinkHandlers = {
+  [PaymentDeepLinkHandler.Disabled]: paymentDeepLinkHandlerDisabled,
+  [PaymentDeepLinkHandler.Merchant]: paymentDeepLinkHandlerMerchant,
+}
+
+function* paymentDeepLinkHandlerDisabled(barcode: QrCode) {
+  yield put(showError(ErrorMessages.QR_FAILED_INVALID_ADDRESS))
+  Logger.error('A payment deep link was scanned without a set handler', barcode.data)
+}
+
+function* paymentDeepLinkHandlerMerchant(barcode: QrCode) {
+  const numberVerified = yield select(numberVerifiedSelector)
+  if (numberVerified || true) {
```

Comment:
Is having a verified number a requirement? Feels like it shouldn't but in any case this `|| true` shouldn't be here I assume? For testing, in the settings menu you can toggle verification status if that's useful

Reply:
Good catch! Indeed I was bypassing it like that locally. I didn't know of the toggle, thanks for that! 👍

Reply:
And yes, the verified number is a requirement I had